instrumentation/screen-orientation/src/main/java/io/opentelemetry/android/instrumentation/screenorientation/ScreenOrientationInstrumentation.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation.screenorientation
+
+import com.google.auto.service.AutoService
+import io.opentelemetry.android.instrumentation.AndroidInstrumentation
+import io.opentelemetry.android.instrumentation.InstallationContext
+
+/**
+ * An Android instrumentation module that installs and manages [ScreenOrientationDetector].
+ */
+@AutoService(AndroidInstrumentation::class)
+class ScreenOrientationInstrumentation : AndroidInstrumentation {
+    private lateinit var detector: ScreenOrientationDetector
+
+    override fun install(ctx: InstallationContext) {
+        val logger =
+            ctx.openTelemetry
+                .logsBridge
+                .loggerBuilder("io.opentelemetry.$name")
+                .build()
+        val applicationContext = ctx.context.applicationContext
+        detector = ScreenOrientationDetector(applicationContext, logger)
+        applicationContext.registerComponentCallbacks(detector)
+    }
+
+    override fun uninstall(ctx: InstallationContext) {
+        if (!::detector.isInitialized) return
```

Comment:
If we need to check whether some field is initialized or not, I prefer to make them nullable instead of resorting to the `isInitialized` check.